storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -202,6 +205,10 @@ void initializeMesosStormConf(Map conf, String localDir) {
 
     _allowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_ALLOWED_HOSTS));
     _disallowedHosts = listIntoSet((List<String>) conf.get(CONF_MESOS_DISALLOWED_HOSTS));
+    Object constraints = conf.get(CONF_MESOS_CONSTRAINTS);
+    if (constraints instanceof Map) {
```

Comment:
if constraints is not `null` and it's not a `Map`, we should throw an `IllegalArgument` exception (i.e., the `mesos.constraints` field was specified, but it wasn't the right type, e.g., maybe it was set to `false`).